docs/spec.md

```diff
@@ -446,18 +435,19 @@ to that of the derivation (or derivations).
 
 
 
-## Traversable
-
-#### Dependencies
-
-  1. Functor
-  1. Foldable
+### Traversable
 
-#### Methods
+In the following signature `Applicative<U>` means that a value must
+not only match `Applicative` signature, but also fully support `Applicative` algebra.
 
-  1. `traverse :: (Traversable t, Applicative f) => Type t ~> (Type f, (a → f b), t a) → f (t b)`
+```js
+Traversable<T> {
+  traverse: (Applicative<U>, a => U<b>, T<a>) => U<T<b>>
+}
+```
```

Comment:
No no no, this is not how this supposed to work. There is a distinction between modules and types that they handle. For example:

```js
const MyModule = {
  map: (fn ,arr) => arr.map(fn)
}
```

Here `MyModule` is a module that supports `Functor` with `T=Array`, and it works with values of type `Array`. In a signature when we see `Functor<Array>` we can pass `MyModule` in that position, but `MyModule<a>` isn't a thing at all, the type `MyModule` works with is `Array<a>`.